bndtools.core/src/bndtools/editor/project/RepositoriesEditModel.java

```diff
@@ -126,7 +127,7 @@ boolean remove(Repository repository) throws Exception {
     }
 
     private String toName(Repository r) {
-        return r.toString();
+        return OSGiRepository.getRepositoryName(r);
```

Comment:
As I mentioned in the bnd PR, OSGiRepository is not a valid place to put this method as it has nothing to do with OSGiRepository.